legacy_code/function_2.py

```diff
@@ -67,7 +75,7 @@ def lambda_handler(event, context):
             table = f.split('.')[-2]
 
             logging.info('Checking if previously uploaded')
-            if (pth, table) in uploaded_files:
+            if "%s-%s" % (md5(pth), table) in uploaded_files:
```

Comment:
Yes, I found that it reduces memory usage from 1.3 GB to 0.7 GB, and add initiating time by 3 seconds. I did the calculation, it greatly reduces the total cost, since we need the memory for all 5 min runtime.